src/redis_lock/__init__.py

```diff
@@ -88,18 +100,37 @@ def id(self):
     def get_owner_id(self):
         return self._client.get(self._name)
 
-    def acquire(self, blocking=True):
+    def acquire(self, blocking=True, timeout=None):
+        """
+        :param blocking:
+            Boolean value specifying whether lock should be blocking or not.
+        :param timeout:
+            An integer value specifying the maximum number of seconds to block.
+        """
         logger.debug("Getting %r ...", self._name)
 
         if self._held:
             raise AlreadyAcquired("Already aquired from this Lock instance.")
 
+        if not blocking and timeout is not None:
+            raise TimeoutNotUsable
+
+        timeout = timeout if timeout is None else int(timeout)
+        if timeout is not None and timeout <= 0:
+            raise InvalidTimeout
+
+        if timeout and self._expire and timeout > self._expire:
+            raise TimeoutIsGreaterThanExpire
```

Comment:
Eg: include the actual values of _expire and timeout so users don't have to start debugging their code to find out what they passed in.